src/lib/FreeC/Backend/Agda/Base.hs

```diff
@@ -43,27 +46,52 @@ imports = Agda.simpleImport $ Agda.qname [baseLibName] $ Agda.name "Free"
 -- Free Monad                                                                --
 -------------------------------------------------------------------------------
 
+-- | Identifier for the @Free@ monad.
 free :: Agda.Name
 free = Agda.name "Free"
 
+-- | Lifts a type in the free monad.
+free' :: Agda.Expr -> Agda.Expr
+free' = Agda.app $ freeArgs $ Agda.qname' free
+
+-- | Apply the @Shape@ and @Position@ argument to the given type constructor.
+--
+-- > c ↦ c @S @P
+freeArgs :: Agda.QName -> Agda.Expr
+freeArgs qname = foldl1
+  Agda.app
+  [ Agda.Ident qname
+  , Agda.Ident (Agda.qname' shape)
+  , Agda.Ident (Agda.qname' position)
+  ]
+
+-- | Identifier for the @Pure@ constructor of the @Free@ monad.
 pureConName :: Agda.Name
 pureConName = Agda.name "pure"
 
+-- | Applies the @Pure@ constructor of the free monad to the given expression.
 pure :: Agda.Expr -> Agda.Expr
 pure = Agda.app $ Agda.Ident $ Agda.qname [Agda.name "Free"] pureConName
 
+-- | Identifier for the @Impure@ constructor of the @Free@ monad.
 impureConName :: Agda.Name
 impureConName = Agda.name "impure"
 
+-- | Applies the @Impure@ constructor of the free monad to the given expression.
 impure :: Agda.Expr -> Agda.Expr
 impure = Agda.app $ Agda.Ident $ Agda.qname [Agda.name "Free"] impureConName
 
+-- | Reserved name for the @Shape@ type variable.
```

Comment:
Somehow I have the feeling that I wanted to rename `shape` and `position` in an earlier version of this file. Did we agree not to do it or was it just forgotten? I wanted to use the same names as in Coq.

Reply:
I think i created this branch before changing the names on the other branch. The variables are named `Shape` and `Position` on the [`dev-agda`](https://github.com/FreeProving/free-compiler/blob/dev-agda/src/lib/FreeC/Backend/Agda/Base.hs) branch. Since the other commit is newer the names from the master branch should persist (even though i am still in favour of `S` and `P`, because they shorten type signatures immensely).